### PR TITLE
The relation walker docs were wrong

### PR DESCRIPTION
**Pull request by pimeys**

### CODE REVIEW

File: libs/datamodel/parser-database/src/walkers/relation/two_way_embedded_many_to_many.rs [hunk 1]
```diff
@@ -5,8 +5,12 @@ use crate::{
...
 use super::RelationWalker;
 
-/// Describes an implicit m:n relation between two models. Neither side defines fields, attributes
-/// or referential actions, which are all inferred by Prisma.
+/// Describes an explicit m:n relation between two models. Both sides define
+/// `fields` which must be a single array scalar field, and `references` that
+/// should point to a single scalar field on the referenced model.
+///
+/// The type of the referenced field must be of the same type but of singular
+/// arity as the referencing field.
```
Comment (tomhoule) [inline libs/datamodel/parser-database/src/walkers/relation/two_way_embedded_many_to_many.rs:13]: I don't get the second part of the sentence ("of singular arity"). But yep, the docs were those for implicit m2m, they were wrong.

Comment (pimeys) [reply libs/datamodel/parser-database/src/walkers/relation/two_way_embedded_many_to_many.rs:13]: How would you write that, so `fields` points to an `Int[]`, the references must point to an `Int`.

Comment (tomhoule) [reply libs/datamodel/parser-database/src/walkers/relation/two_way_embedded_many_to_many.rs:13]: Aah, `the referencing field must be a list field, the referenced single id field must be a required field`.

Comment (pimeys) [reply libs/datamodel/parser-database/src/walkers/relation/two_way_embedded_many_to_many.rs:13]: List field and required field of the same type. Although `Int[]` and `Int` are different types, so I don't know the best way to write this.

Comment (tomhoule) [reply libs/datamodel/parser-database/src/walkers/relation/two_way_embedded_many_to_many.rs:13]: Or we could just omit that sentence? :p

**Pull request merged**